catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/notes/NoteUtil.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.notes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.boon.json.JsonFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Core;
+
+public class NoteUtil {
+
+    private NoteUtil() { }
+
+    /**
+     * Extracts the data required by the user from the note {@link Metacard} object.
+     * If any required fields are null the return object will also be null.
+     *
+     * @param metacard - the metacard object to abstract response data from
+     * @return map of the response note fields
+     **/
+    public static Map<String, String> getResponseNote(Metacard metacard) {
```

Comment:
❗️ Should annotate this method with `@Nullable` and also clarify the `@return` javadoc to state it can return null
❓ Would `EndpointUtil::getMetacardMap` serve a similiar enough purpose to this method it could be reused?

In the future it might be nice to create a transformer for this like we use for the other metacards that go through the backend, but thats a little outside the scope of this.

Reply:
I'll add the `@Nullable`, but I don't see how it could be merged with `getMetacardMap`.